src/jobservice/job/impl/notification/slack_job.go

```diff
@@ -0,0 +1,104 @@
+package notification
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+	"os"
+	"strconv"
+
+	"github.com/goharbor/harbor/src/jobservice/job"
+	"github.com/goharbor/harbor/src/jobservice/logger"
+	"time"
+)
+
+// SlackJob implements the job interface, which send notification to slack by slack incoming webhooks.
+type SlackJob struct {
+	client *http.Client
+	logger logger.Interface
+}
+
+// MaxFails returns that how many times this job can fail.
+func (sj *SlackJob) MaxFails() (result uint) {
+	// Default max fails count is 10, and its max retry interval is around 3h
+	// Large enough to ensure most situations can notify successfully
+	result = 10
+	if maxFails, exist := os.LookupEnv(maxFails); exist {
+		mf, err := strconv.ParseUint(maxFails, 10, 32)
+		if err != nil {
+			logger.Warningf("Fetch slack job maxFails error: %s", err.Error())
+			return result
+		}
+		result = uint(mf)
+	}
+	return result
+}
+
+// MaxCurrency is implementation of same method in Interface.
+func (sj *SlackJob) MaxCurrency() uint {
+	return 1
+}
+
+// ShouldRetry ...
+func (sj *SlackJob) ShouldRetry() bool {
+	return true
+}
+
+// Validate implements the interface in job/Interface
+func (sj *SlackJob) Validate(params job.Parameters) error {
```

Comment:
I saw this job has parameters. I'd like to suggest validating the parameters here.

Reply:
ok. i will check.

Reply:
i checked that all jobs have parameters but leave the validate function null. so go through it in future?

Reply:
fixed